viewing_party/party.py

```diff
@@ -1,23 +1,225 @@
+        genere_dict[user_data["watched"][i]["genre"]] = genere_dict.get(user_data["watched"][i]["genre"],0) + 1
+    
+    max_genere = ["genere", -1000]
+
+    for genere_key, genere_value in genere_dict.items():
+        if genere_value > max_genere[1]:
+            max_genere = [genere_key, genere_value]
+
+    return max_genere[0]
+    # genre_count = 0
+    # most_popular_genre = " "
+    # for genere_key, genere_value in genere_dict.items():
+    #     if genere_value > genre_count:
+    #         most_popular_genre = genere_key
+    #         genre_count = genere_value
+
+    # return most_popular_genre
+
+
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+
+# # USER_DATA_2b = {
+#     "watched": [
+#         INTRIGUE_1,
+#         FANTASY_2,
+#         ACTION_1,
+#         {
+#     "title": "The Lord of the Functions: The Fellowship of the Function",
+#     "genre": "Fantasy",
+#     "rating": 4.8
+# },
+#         FANTASY_3,
+#         INTRIGUE_2,
+#     ],
+#     "friends": [
+#         {
+#             "watched": [
+#                 { "title": "The Lord of the Functions: The Fellowship of the Function",
+#                      "genre": "Fantasy",
+#                         "rating": 4.8, 
+#                           "host" = "netflix"},
+#                 FANTASY_3,
+#                 FANTASY_4,
+#                 HORROR_1,
+#             ]
+#         },
+#         {
+#             "watched": [
+#                 FANTASY_1,
+#                 ACTION_1,
+#                 INTRIGUE_1,
+#                 INTRIGUE_3,
+#             ]
+#         }
+#     ]  
+# }
+
+# def get_unique_watched(user_data):
+#     unique_watched = []
+#     duplicate = False
+#     for i in range(len(user_data["watched"])):
+#         duplicate = False
+#         # if item is in set(t1, t2, t3):
+#         for n in range(len(user_data["friends"])):
+#             for m in range(len(user_data["friends"][n]["watched"])):
+#                 if user_data["watched"][i]["title"] == user_data["friends"][n]["watched"][m]["title"]:
+#                     duplicate = True
+#                     break
+#             if duplicate:
+#                 break
+#         if not duplicate: 
+#             unique_watched.append(user_data["watched"][i])
+
+#     # if  user_data["watched"][i]["title"] != user_data["friends"][n]["watched"][m]["title"]:    
+#     return   unique_watched
+def get_title_friends_watched(user_data):
+    movie_titile_friends_watched = set()
+
+    for d in user_data["friends"]:
+        for d_movie in d["watched"]:
+            movie_titile_friends_watched.add(d_movie["title"])
+
+    return movie_titile_friends_watched
+
+def get_unique_watched(user_data):
+    unique_watched = []
+
+    for movie in (user_data["watched"]):
+        if movie["title"] not in get_title_friends_watched(user_data):
+            unique_watched.append(movie)
+
+    return   unique_watched

```

Comment:
Nice work here! Your helper function helps us focus on the actual purpose of the function and using a `set` lets your membership testing be done in `O(1)` constant time! 
```suggestion
    unique_watched = []

    for movie in (user_data["watched"]):
        if movie["title"] not in get_title_friends_watched(user_data):
            unique_watched.append(movie)

    return  unique_watched
```